### PR TITLE
[NOD-1168] Add context interfaces for flows

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: protocol/flows/ping/ping.go [hunk 1]
```diff
@@ -11,9 +11,13 @@ import (
 	"github.com/kaspanet/kaspad/wire"
 )
 
+// Context is the interface for the context needed for the Ping and Pong flows.
```
Comment (svarogg) [inline protocol/flows/ping/ping.go:14]: Maybe no need for Context if nothing in there?
Still might be good as a placeholder, not sure.

Anyway, if decide to keep it, please rename as in all other places [FlowName]Context

File: protocol/flowcontext/flow_context.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	addressManager *addrmgr.AddrManager, txPool *mempool.TxPool, netAdapter *netadapter.NetAdapter) *FlowContext {
+	return &FlowContext{
+		cfg:                         cfg,
+		netAdapter:                  netAdapter,
+		dag:                         dag,
+		addressManager:              addressManager,
+		txPool:                      txPool,
+		sharedRequestedTransactions: relaytransactions.NewSharedRequestedTransactions(),
+		sharedRequestedBlocks:       blockrelay.NewSharedRequestedBlocks(),
+	}
```
Comment (stasatdaglabs) [inline protocol/flowcontext/flow_context.go:54]: readyPeers needs to be initialized here as well.

Comment (stasatdaglabs) [reply protocol/flowcontext/flow_context.go:54]: Likewise transactionsToRebroadcast.

File: protocol/flows/relaytransactions/shared_requested_transactions.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ func (s *SharedRequestedTransactions) addIfNotExists(txID *daghash.TxID) (exists
 	return false
 }
 
-// NewSharedRequestedTransactions returns a new instance of SharedRequestedTransactions.
+// NewSharedRequestedTransactions returns a new instance of *SharedRequestedTransactions.
```
Comment (stasatdaglabs) [inline protocol/flows/relaytransactions/shared_requested_transactions.go]: Whether you return it by reference or by value is probably redundant to specify in a comment.

File: protocol/flows/handshake/handshake.go [hunk 4]
```diff
@@ -5,21 +5,30 @@ import (
...
 	routerpkg "github.com/kaspanet/kaspad/netadapter/router"
-	"github.com/kaspanet/kaspad/protocol/flows/ibd"
 	peerpkg "github.com/kaspanet/kaspad/protocol/peer"
 	"github.com/kaspanet/kaspad/util/locks"
 	"github.com/kaspanet/kaspad/wire"
 	"github.com/pkg/errors"
 )
 
+// Context is the interface for the context needed for the HandleHandshake flow.
+type Context interface {
```
Comment (stasatdaglabs) [inline protocol/flows/handshake/handshake.go]: Please rename this to HandshakeContext.

**Pull request merged**